packages/smart-wallet/src/offers.js

```diff
@@ -71,19 +71,28 @@ export const makeOfferExecutor = ({
         status = { ...status, ...changes };
         onStatusChange(status);
       };
+
       /**
        * Notify user and attempt to recover
        *
        * @param {Error} err
+       * @param {UserSeat<unknown>?} seatRef
        */
-      const handleError = err => {
+      const handleError = (err, seatRef) => {
```

Comment:
this optional seatRef is a little loose. Instead leave this `handleError` alone and make a new function like` `exitSeatAndHandleError` in the `tryBody` scope. It won't need to take `seatRef` because it's in the closure. Then the error handler lines can continue to be a prebound function instead of a new one for each `when`.

Also better to exit the offer before reclaiming payments. I don't know if it affects anything in practice but it's conceptually more clear.